robusta-codegen/src/transformation/imported.rs

```diff
@@ -130,19 +130,38 @@ impl<'ctx> Fold for ImportedMethodTransformer<'ctx> {
 
                 let input_types_conversions = signature
                     .inputs
-                    .iter()
+                    .iter_mut()
                     .filter_map(|i| match i {
                         FnArg::Typed(t) => match &*t.pat {
                             Pat::Ident(PatIdent { ident, .. }) if ident == "self" => None,
-                            _ => Some((&t.ty, t.ty.span()))
+                            _ => Some((&t.ty, t.ty.span(), &mut t.attrs))
                         },
                         FnArg::Receiver(_) => None,
                     })
-                    .map(|(t, span)| {
-                        if let CallType::Safe(_) = call_type {
-                            quote_spanned! { span => <#t as ::robusta_jni::convert::TryIntoJavaValue>::SIG_TYPE, }
+                    .map(|(t, span, attrs)| {
+                        let override_input_type = attrs.iter().next().and_then(|attr| {
+                            if attr.path.segments.iter().find(|seg| seg.ident.to_string().as_str() == "input_type").is_some() {
+                                let token_tree: Group = syn::parse2::<Group>(attr.clone().tokens).unwrap();
+                                let token_tree_lit: Lit = syn::parse2::<Lit>(token_tree.stream()).unwrap();
+
+                                if let Lit::Str(literal) = token_tree_lit {
+                                    Some(literal.value())
+                                } else {
+                                    None
+                                }
+                            } else {
+                                None
+                            }
+                        });
```

Comment:
- Maybe with `darling` we can simplify this a bit.
- I think it would be better to use `attrs.find()` instead of `attrs.next()`, as the signature could have potentially more than one attribute attached to it.